src/component.ts

```diff
@@ -1,4 +1,6 @@
+import 'reflect-metadata'
```

Comment:
I'm not a fan of adding dependency that some users may not need as it bloats the lib size. Can we remove the dep and make this feature be optional?
I think only forwarding metadata when Reflect metadata is supported would work.

Reply:
That's reasonable, I'll fix it.